src/main/java/com/intel/gkl/compression/IntelDeflater.java

```diff
@@ -171,15 +179,15 @@ public void finish() {
      *         output buffer
      */
     @Override
-    public int deflate(byte[] b, int off, int len )  throws NullPointerException, IllegalArgumentException {
+    public int deflate(byte[] b, int off, int len) {
```

Comment:
I think this function should have a "thows" statement of the few types it can throws.  Is there some reason we do not do it here?